src/components/PlaylistTable.jsx

```diff
@@ -0,0 +1,139 @@
+      if (firstPage) {
+        return $.when.apply($, [
+          apiCall(
+            "https://api.spotify.com/v1/users/" + userId + "/tracks",
+            this.props.access_token
+          ),
+          apiCall(
+            "https://api.spotify.com/v1/users/" + userId + "/playlists",
+            this.props.access_token
+          )
+        ])
+      } else {
+        return apiCall(url, this.props.access_token);
+      }
+    }.bind(this)).done(function() {
+      var response;
+      var playlists = [];
+
+      if (arguments[1] === 'success') {
+        response = arguments[0];
+        playlists = arguments[0].items;
+      } else {
+        response = arguments[1][0];
+        playlists = arguments[1][0].items;
+      }
+
+      // Show library of saved tracks if viewing first page
+      if (firstPage) {
+        playlists.unshift({
+          "id": "liked",
+          "name": "Liked",
+          "public": false,
+          "collaborative": false,
+          "owner": {
+            "id": userId,
+            "display_name": userId,
+            "uri": "spotify:user:" + userId
+          },
+          "tracks": {
+            "href": "https://api.spotify.com/v1/me/tracks",
+            "limit": arguments[0][0].limit,
+            "total": arguments[0][0].total
+          },
+          "uri": "spotify:user:" + userId + ":saved"
+        });
+
+        // FIXME: Handle unmounting
+        this.setState({
+          likedSongsLimit: arguments[0][0].limit,
+          likedSongsCount: arguments[0][0].total
+        })
+      }
+
+      // FIXME: Handle unmounting
+      this.setState({
+        playlists: playlists,
+        playlistCount: response.total,
+        nextURL: response.next,
+        prevURL: response.previous
+      });
+
+      $('#playlists').fadeIn();
+      $('#subtitle').text((response.offset + 1) + '-' + (response.offset + response.items.length) + ' of ' + response.total + ' playlists for ' + userId)
+
+    }.bind(this))
+  }
+
+  exportPlaylists() {
+    PlaylistsExporter.export(this.props.access_token, this.state.playlistCount, this.state.likedSongsLimit, this.state.likedSongsCount);
+  }
+
+  componentDidMount() {
+    this.loadPlaylists(this.props.url);
+  }
+
+  render() {
+    if (this.state.playlists.length > 0) {
+      return (
+        <div id="playlists">
+          <Paginator nextURL={this.state.nextURL} prevURL={this.state.prevURL} loadPlaylists={this.loadPlaylists.bind(this)}/>
+          <table className="table table-hover">
+            <thead>
+              <tr>
+                <th style={{width: "30px"}}></th>
+                <th>Name</th>
+                <th style={{width: "150px"}}>Owner</th>
+                <th style={{width: "100px"}}>Tracks</th>
+                <th style={{width: "120px"}}>Public?</th>
+                <th style={{width: "120px"}}>Collaborative?</th>
+                <th style={{width: "100px"}} className="text-right">
```

Comment:
you should define your width using css classes.
Directly defining the style object is required when using variable / computing the value.
Aside of separating the visual from your logic, doing so won't break your Jest snapshots if you decide to change a column width

Reply:
Very good point about the Jest snapshots! Indeed, something else to add to the cleanup list 👍